TombEngine/Shaders/Rooms.fx

```diff
@@ -83,17 +83,7 @@ PixelShaderInput VS(VertexShaderInput input)
 	output.Normal = input.Normal;
 	output.Color = float4(col, input.Color.w);
 	output.PositionCopy = screenPos;
-
-#ifdef ANIMATED
-
-	if (Type == 0)
-		output.UV = GetFrame(input.PolyIndex, input.AnimationFrameOffset);
-	else
-		output.UV = input.UV; // TODO: true UVRotate in future?
-#else
-	output.UV = input.UV;
-#endif
-	
+    output.UV = GetUVPossiblyAnimated(input.UV, input.PolyIndex, input.AnimationFrameOffset);

```

Comment:
Can we profile performance with animated textures on? I see that `GetUVPossiblyAnimated` has an if condition. It may potentially lead to big perfo issues.

Reply:
Branching is done via a constant buffer variable and not due to the result of a previous elaboration so the GPU should optimize it.